cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -0,0 +1,114 @@
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	minishiftConfig "github.com/minishift/minishift/pkg/minishift/config"
+	"github.com/minishift/minishift/pkg/minishift/oc"
+	profileActions "github.com/minishift/minishift/pkg/minishift/profile"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+const (
+	emptyProfileError  = "You must provide the profile name. Run `minishift profile list` to view profiles"
+	extraArgumentError = "You have provided more arguments than required. You must provide a single profile name"
+)
+
+var profileSetCmd = &cobra.Command{
+	Use:   "set",
+	Short: "Sets the default profile for Minishift",
+	Long:  "Sets the default profile for Minishift. After you set the profile, all commands will use the profile by default",
+	Run: func(cmd *cobra.Command, args []string) {
+		var doesProfileExist = false
+		if len(args) == 0 {
+			atexit.ExitWithMessage(1, emptyProfileError)
+		} else if len(args) > 1 {
+			atexit.ExitWithMessage(1, extraArgumentError)
+		}
+
+		profileName := args[0]
+
+		//check if the profile is present in the AllInstancesConfig
+		profileList := profileActions.GetProfileNameList()
+		for i := range profileList {
+			if profileList[i] == profileName {
+				doesProfileExist = true
+			}
+		}
+		if !doesProfileExist {
+			atexit.ExitWithMessage(1, fmt.Sprintf("Error: '%s' is not a valid profile", profileName))
+		}
+
+		//Check if the requested profile is already active
+		if profileName == profileActions.GetActiveProfile() {
+			atexit.ExitWithMessage(1, fmt.Sprintf("'%s' is already the active profile", profileName))
+		}
+
+		err := profileActions.SetActiveProfile(profileName)
+		if err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		} else {
+			fixOcCliContext(profileName)
+			fmt.Println(fmt.Sprintf("Profile set to '%s' successfully", profileName))
+		}
+
+	},
+}
+
+func fixOcCliContext(profileName string) {
+	const (
+		defaultProject = "myproject"
+		defaultUser    = "developer"
+	) //This needs to be fixed
+
+	//We need to reassign ProfileName, MachineName, KubeConfigPath as it would have values for the previous
+	// profile
+	constants.ProfileName = profileName
+	constants.MachineName = constants.ProfileName
+	constants.KubeConfigPath = filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig")
+
+	api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+	defer api.Close()
+
+	host, err := api.Load(constants.MachineName)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+	ip, err := host.Driver.GetIP()
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error getting the IP address: %s", err.Error()))
+	}
+
+	ocRunner, err := oc.NewOcRunner(minishiftConfig.InstanceConfig.OcPath, constants.KubeConfigPath)
```

Comment:
I think this should be shared code...

what is actual happening? it logs in and sets the default project?